### ISSUE TITLE
How do I write a 403 to the client?

### ISSUE THREAD

**Issue by herebebeasties** [justcoding121/Titanium-Web-Proxy#106]
There's a `SessionEventArgs.Ok(...)` which writes 200 OK responses, but I don't want that - I want my proxy to be able to block things, and have the clients be able to detect that. It shouldn't say the request has made it through fine when it hasn't.

I can create my own `Response` object and set the `ResponseStatusCode` manually and send it, but then I can't set the body to something useful because `ResponseBody` is internal. The RFC says clients can ignore the `ResponseStatusDescription`, and many do.

Additionally, I can't call set the `CancelRequest` field on the `Request` object if I try to do this stuff manually, either, so I get a round-trip even though I don't want one.

In short, I can't see a way to do this without patching the library to add `SessionEventArgs.CustomReturnCode(...)` type method.

Am I missing something?

**Comment by justcoding121** [comment]
We can add the feature. If you have already written a fix feel free for a PR.

**Comment by kevinrmcd** [comment]
For a private release, add a new class under \http\Reponses\GenericResponse.cs, with this content:
```
using System.Net;

namespace Titanium.Web.Proxy.Http.Responses
{
    /// <summary>
    /// Anything other than a 200 or 302 response
    /// </summary>
    public class GenericResponse : Response
    {
        public GenericResponse(HttpStatusCode status)
        {
            ResponseStatusCode = ((int)status).ToString();
            ResponseStatusDescription = status.ToString(); 
        }

        public GenericResponse(string statusCode, string statusDescription)
        {
            ResponseStatusCode = statusCode;
            ResponseStatusDescription = statusCode;
        }
    }
}
```
And then, in SessionEventArgs.cs, add this:
```
public async Task GenericResponse(string html, HttpStatusCode status)
{
	await GenericResponse(html, null, status);
}

public async Task GenericResponse(string html, Dictionary<string, HttpHeader> headers, HttpStatusCode status)
{
	if (WebSession.Request.RequestLocked)
	{
		throw new Exception("You cannot call this function after request is made to server.");
	}

	if (html == null)
	{
		html = string.Empty;
	}

	var result = Encoding.Default.GetBytes(html);

	await GenericResponse(result, headers, status);
}

public async Task GenericResponse(byte[] result, Dictionary<string, HttpHeader> headers, HttpStatusCode status)
{
	var response = new GenericResponse(status);

	if (headers != null && headers.Count > 0)
	{
		response.ResponseHeaders = headers;
	}
	response.HttpVersion = WebSession.Request.HttpVersion;
	response.ResponseBody = result;

	await Respond(response);

	WebSession.Request.CancelRequest = true;
}
```
Then in OnRequest, call like this:
```
await e.GenericResponse("<HTML>Server error</HTML>", HttpStatusCode.500);
```
Of course, if we add this, should probably derive OK from GenericResponse.

**Comment by kevinrmcd** [comment]
I've added this code, just waiting for it to be merged.

**Comment by justcoding121** [comment]
Good work. @kevinrmcd Thanks
